pkg/server/plugin/bundlepublisher/awsrolesanywhere/awsrolesanywhere_test.go

```diff
@@ -0,0 +1,376 @@
+				TrustAnchorID:   "trust-anchor-id",
+			},
+			expectCode: codes.InvalidArgument,
+			expectMsg:  "missing bundle in request",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			var err error
+			options := []plugintest.Option{
+				plugintest.CaptureConfigureError(&err),
+				plugintest.CoreConfig(catalog.CoreConfig{
+					TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
+				}),
+				plugintest.ConfigureJSON(tt.config),
+			}
+
+			newClient := func(awsConfig aws.Config) (rolesAnywhere, error) {
+				mockClient := fakeClient{
+					t:                    t,
+					expectTrustAnchorID:  aws.String(tt.config.TrustAnchorID),
+					updateTrustAnchorErr: tt.updateTrustAnchorErr,
+				}
+				return &mockClient, nil
+			}
+			p := newPlugin(newClient)
+
+			if tt.config != nil {
+				plugintest.Load(t, builtin(p), nil, options...)
+				require.NoError(t, err)
+			}
+
+			resp, err := p.PublishBundle(context.Background(), &bundlepublisherv1.PublishBundleRequest{
+				Bundle: tt.bundle,
+			})
+
+			if tt.expectMsg != "" {
+				spiretest.RequireGRPCStatusContains(t, err, tt.expectCode, tt.expectMsg)
+				return
+			}
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+		})
+	}
+}
+
+func TestPublishMultiple(t *testing.T) {
+	config := &Config{
+		AccessKeyID:     "access-key-id",
+		SecretAccessKey: "secret-access-key",
+		Region:          "region",
+		TrustAnchorID:   "trust-anchor-id",
+	}
+
+	var err error
+	options := []plugintest.Option{
+		plugintest.CaptureConfigureError(&err),
+		plugintest.CoreConfig(catalog.CoreConfig{
+			TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
+		}),
+		plugintest.ConfigureJSON(config),
+	}
+
+	newClient := func(awsConfig aws.Config) (rolesAnywhere, error) {
+		return &fakeClient{
+			t:                   t,
+			expectTrustAnchorID: aws.String(config.TrustAnchorID),
+		}, nil
+	}
+	p := newPlugin(newClient)
+	plugintest.Load(t, builtin(p), nil, options...)
+	require.NoError(t, err)
+
+	// Test multiple update trust anchor operations, and check that only a call to
+	// UpdateTrustAnchor is made when there is a modified bundle that was not successfully
+	// published before.
+
+	// Have an initial bundle with SequenceNumber = 1.
+	bundle := getTestBundle(t)
+	bundle.SequenceNumber = 1
+
+	client, ok := p.rolesAnywhereClient.(*fakeClient)
+	require.True(t, ok)
+
+	// Reset the API call counters.
+	client.updateTrustAnchorCount = 0
+
+	// Throw an error when calling UpdateTrustAnchor.
+	client.updateTrustAnchorErr = errors.New("error calling UpdateTrustAnchor")
+
+	// Call PublishBundle. UpdateTrustAnchor should be called and return an error.
+	resp, err := p.PublishBundle(context.Background(), &bundlepublisherv1.PublishBundleRequest{
+		Bundle: bundle,
+	})
+	require.Error(t, err)
+	require.Nil(t, resp)
+
+	// UpdateTrustAnchor was called, even though it failed, so its counter should be incremented.
+	require.Equal(t, 1, client.updateTrustAnchorCount)
```

Comment:
This should remain in 0 due to the error.
```suggestion
	require.Equal(t, 0, client.updateTrustAnchorCount)
```